src/steps/fetchServices.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  IntegrationStep,
+  IntegrationStepExecutionContext,
+  createIntegrationEntity,
+  createIntegrationRelationship,
+  JobState,
+} from '@jupiterone/integration-sdk';
+import _ from 'lodash';
+
+import { OnCall, Service } from '../types';
+import { requestAll } from '../pager-duty';
+import { reduceGroupById } from '../utils/reduceGroupById';
+
+const step: IntegrationStep = {
+  id: 'fetch-services',
+  name: 'Fetch Services',
+  dependsOn: ['fetch-teams', 'fetch-users'],
+  types: ['pagerduty_service'],
+  async executionHandler({
+    logger,
+    jobState,
+    instance,
+  }: IntegrationStepExecutionContext) {
+    logger.info('Requesting /services endpoint');
+    const { apiKey } = instance.config;
+    const services = await requestAll<Service>('/services', 'services', apiKey);
+
+    await buildServiceEntities(jobState, services);
+
+    await buildTeamRelations(jobState, services);
+
+    logger.info('Requesting /oncalls endpoint');
+    const oncalls = await requestAll<OnCall>('/oncalls', 'oncalls', apiKey);
+
+    await buildOnCallRelations(jobState, oncalls, services);
+  },
+};
+
+async function buildServiceEntities(
+  jobState: JobState,
+  services: Service[],
+): Promise<void> {
+  const serviceEntities = services.map((service) =>
+    createIntegrationEntity({
+      entityData: {
+        source: {
+          id: service.id,
+          name: service.name,
+          description: service.description,
+          summary: service.summary,
+        },
```

Comment:
The `source` is meant to be the complete provider representation of the service. It will include lots of data that we don't ingest into the graph, stored as raw data in S3 and visible in the UI.

This `createIntegrationEntity` definitely needs more documentation. I've created https://github.com/JupiterOne/integration-sdk/issues/77